### PR TITLE
improvement(sct.py): add support to list azure images

### PR DESCRIPTION
**Pull request by fgelcer**
we now sync Azure images to multiple regions,
so now we added support to running Azure tests in multiple regions.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sct.py [hunk 1]
```diff
@@ -612,7 +613,21 @@ def list_images(cloud_provider: str, branch: str, version: str, region: str, arc
...
                 )
             case "azure":
-                click.echo("Azure image listing not implemented yet.")
+                if arch:
+                    click.echo("WARNING:--arch option not implemented currently for Azure machine images.")
+                azure_images = azure_utils.get_scylla_images(scylla_version=version, region_name=region)
+                rows = list()
+                for image in azure_images:
+                    rows.append(['Azure', image.name, image.id, 'N/A'])
+                # rows = [['Azure', image.name, image.id, image.tags.get('creation_date', "N/A")]
```
Comment (soyacz) [inline sct.py:622]: remove comment

**Review by soyacz — APPROVED** [review]
besides redundant comment. LGTM.

**Comment by fgelcer** [comment]
i just figured out that i implemented only half of it... 
i need still to implemente the `branch` part of the command
will work on it later this week, so please, DON"T MERGE IT YET

File: sct.py (hunk 1 above)
Comment (fgelcer) [reply sct.py]: fixed

**Comment by fgelcer** [comment]
i have implemented now for both by `branch` and by `version`

**Comment by fgelcer** [comment]
@soyacz @Mark-Gurevich , please review.

i have tested it locally, and it looks like:
```
$ hydra list-images -c azure -r eastus -br branch-2023.1:latest
WARNING:--arch option not implemented currently for Azure machine images.
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                  Azure Machine Images by version                                                                                                                   |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
| Backend | Name                                                      | ImageId                                                                                                                                                                       | CreationDate |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
| Azure   | scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-05T10-21-36 | /subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-05T10-21-36 | N/A          |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
```

and:
```
$ hydra list-images -c azure -r eastus -v 2023.1.0-rc2
WARNING:--arch option not implemented currently for Azure machine images.
+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                  Azure Machine Images by version                                                                                                                   |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
| Backend | Name                                                      | ImageId                                                                                                                                                                       | CreationDate |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
| Azure   | scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-02T14-20-58 | /subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-02T14-20-58 | N/A          |
| Azure   | scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-02T18-22-21 | /subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-02T18-22-21 | N/A          |
| Azure   | scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-05T10-21-36 | /subscriptions/6c268694-47ab-43ab-b306-3c5514bc4112/resourceGroups/SCYLLA-IMAGES/providers/Microsoft.Compute/images/scylla-enterprise-2023.1.0-rc2-x86_64-2023-03-05T10-21-36 | N/A          |
+---------+-----------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------------+
```

**Comment by fgelcer** [comment]
fixed pre-commit checks that were failing.

File: sdcm/utils/version_utils.py [hunk 2]
```diff
@@ -69,7 +69,7 @@
 )
 
 SCYLLA_VERSION_RE = re.compile(r"\d+(\.\d+)?\.[\d\w]+([.~][\d\w]+)?")
-SCYLLA_VERSION_GROUPED_RE = re.compile(r'(?P<version>[\w.~]+)-0?\.?(?P<date>[\d]+)\.(?P<commit_id>\w+)')
+SCYLLA_VERSION_GROUPED_RE = re.compile(r'(?P<version>[\w.~]+)-(?P<build>0|rc\d)?\.?(?P<date>[\d]+)\.(?P<commit_id>\w+)')
```
Comment (soyacz) [inline sdcm/utils/version_utils.py:72]: would be nice to add a case to unit test for it `test_version_utilis.py`

**Review by soyacz — APPROVED** [review]
LGTM

File: sdcm/utils/version_utils.py (hunk 2 above)
Comment (fgelcer) [reply sdcm/utils/version_utils.py:72]: i don't have much resources to do it now... would you mind opening a task for it?

**Review by Mark-Gurevich — APPROVED** [review]

**Comment by fgelcer** [comment]
@roydahan , we need this ASAP on 2023.1 and 5.2... please decide with @scylladb/qa-maintainers if this should be cherry picked or FF'ed

**Comment by roydahan** [comment]
I think that all fixes in master are relevant for 5.2 and 2023.1.
The SLA nemesis need to be carefully looked at.

**Comment by fgelcer** [comment]
> I think that all fixes in master are relevant for 5.2 and 2023.1. The SLA nemesis need to be carefully looked at.

@fruch said they are limited to SLA and it is safe to backport

**Pull request merged**